include/sparrow/buffer.hpp

```diff
@@ -226,11 +289,29 @@ namespace sparrow
         if (n != size())
         {
             buffer<T> tmp(n);
-            std::copy(data(), data() + size(), tmp.data());
+            size_type copy_size = std::min(size(), n);
```

Comment:
Is this for some exception safety?

Reply:
This is to handle resizing with a smaller size.